third_party/move/move-compiler/src/expansion/ast.rs

```diff
@@ -548,6 +548,44 @@ pub enum Exp_ {
 }
 pub type Exp = Spanned<Exp_>;
 
+impl Exp_ {
+    /// Get all names from an expression
+    /// only recursively check on expression types supported in constant definition.
+    pub fn get_names_for_const_exp(&self) -> BTreeSet<Name> {
+        let mut names = BTreeSet::new();
+        let mut add_names = |v: &Exp| {
+            let set = v.value.get_names_for_const_exp();
+            for n in set.iter() {
+                names.insert(*n);
+            }
+        };
+        match self {
+            Self::Name(access, _) => {
+                names.insert(*access.value.get_name());
+            },
+            Self::Call(_, _, _, exp_vec) | Self::Vector(_, _, exp_vec) => {
+                let _ = exp_vec.value.iter().map(&mut add_names);
+            },
+            Self::UnaryExp(_, exp) => {
+                add_names(exp);
+            },
+            Self::BinopExp(exp1, _, exp2) => {
+                add_names(exp1);
+                add_names(exp2);
+            },
+            Self::Block(seq) => {
+                for s in seq.iter() {
+                    if let SequenceItem_::Seq(exp) = &s.value {
```

Comment:
Interesting that we allow sequences in const defintions. Is there any use case for this? Should it be disallowed?

Can this sequence contain local bindings that could clash with the const names?